File: D108/src/main/java/com/example/model/service/RoomService.java
package com.example.model.service;

import com.example.model.dto.RoomDto;
import com.example.model.dto.RoomListDto;

import java.util.List;

public interface RoomService {

    /** 방을 생성한다.**/
    public int createRoom(RoomDto roomDto);

    /** roomId에 해당하는 방 정보를 변경한다.**/
    public int updateRoom(int roomId, RoomDto roomDto);

    /** 전체 방 정보를 조회합니다.**/
    public List<RoomListDto> searchRoom();

    /** 참가자 증가**/
    public void incrementUserCount(int roomId);

    /** 참가자 감소**/
    public void decrementUserCount(int roomId);

    /** 참가자 조회**/
    public int getUserCount(int roomId);
}

File: FE/app/src/main/java/com/example/gametset/room/data/remote/RoomService.kt
package com.example.gametset.room.data.remote

import com.example.gametset.room.data.model.dto.OneRoomDto
import com.example.gametset.room.data.model.dto.RoomChangeDto
import com.example.gametset.room.data.model.dto.RoomCreateDto
import com.example.gametset.room.data.model.dto.RoomDto
import com.example.gametset.room.data.model.response.UserResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface RoomService {

    @GET("room/list")
    suspend fun getAllRoom(): List<OneRoomDto>

    @POST("room")
    suspend fun createRoom(@Body RoomCreateDto: RoomCreateDto): Int

    @PUT("room/{roomId}")
    suspend fun changeRoomInfo(@Path("roomId") roomId:Int, @Body roomChangeDto : RoomChangeDto) : Boolean

    @GET("room/{roomId}")
    suspend fun getOneRoom(@Path("roomId") roomId: Int): OneRoomDto
}
